Source/Core/DolphinWX/Config/GeneralConfigPane.cpp

```diff
@@ -198,7 +199,8 @@ void GeneralConfigPane::OnDualCoreCheckBoxChanged(wxCommandEvent& event)
   if (Core::IsRunning())
     return;
 
-  SConfig::GetInstance().bCPUThread = m_dual_core_checkbox->IsChecked();
+  Config::SetBase(Config::CPU_THREAD, m_dual_core_checkbox->IsChecked());
+  Config::InvokeConfigChangedCallbacks();
```

Comment:
Should this be triggered elsewhere perhaps (so we can't forget to do it?)

Might be a little late to the party, but now that I see the Callback mechanism in action it might be a bit too global, isn't it? On one hand, toggling just one setting like the CPU Core need not reload everything (raise a callback on that particular `ConfigInfo` instead of globally?); but on the other hand it might be nicer to batch operations so callbacks aren't invoked all over the place (perhaps using a RAII guard around places where batch-changes occur? Block/Defer all `ConfigInfo` callbacks and only send one for the whole config instead - given at least one of them changed; or perhaps if a certain threshold is reached where one "everything changed" event is better than _n_ "setting _X_ changed"). Then again, not sure if we even have places where the settings are set in a batch-like fashion; stuff called from GUI is a one-shot, single property thing in most cases.

Just some food for thought, not necessarily a request for changing something. Might turn out that it doesn't matter in practise because its not called often enough to have an impact.